docs/book/src/migration/plugin/v2_v3.md

```diff
@@ -1,3 +1,564 @@
+	go mod init tmp ;\
+	go get sigs.k8s.io/kustomize/kustomize/v3@v3.5.4 ;\
+	rm -rf $$KUSTOMIZE_GEN_TMP_DIR ;\
+	}
+KUSTOMIZE=$(GOBIN)/kustomize
+else
+KUSTOMIZE=$(shell which kustomize)
+endif
+
+```
+
+With:
+```
+# Download controller-gen locally if necessary
+CONTROLLER_GEN = $(shell pwd)/bin/controller-gen
+controller-gen:
+	$(call go-get-tool,$(CONTROLLER_GEN),sigs.k8s.io/controller-tools/cmd/controller-gen@v0.4.1)
+
+# Download kustomize locally if necessary
+KUSTOMIZE = $(shell pwd)/bin/kustomize
+kustomize:
+	$(call go-get-tool,$(KUSTOMIZE),sigs.k8s.io/kustomize/kustomize/v3@v3.8.7)
+
+# go-get-tool will 'go get' any package $2 and install it to $1.
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+define go-get-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
+rm -rf $$TMP_DIR ;\
+}
+endef
+
+```
+### Update your controllers
+
+<aside class="note warning">
+<h1>Note</h1>
+
+The version of `sigs.k8s.io/controller-runtime`. Check [sigs.k8s.io/controller-runtime release docs from 0.7.0+ version][controller-releases] for breaking changes.
+
+</aside>
+
+Replace:
+
+```go 
+func (r *<MyKind>Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+    ctx := context.Background() 
+	log := r.Log.WithValues("cronjob", req.NamespacedName)
+```
+
+With:
+
+```go 
+func (r *<MyKind>Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("cronjob", req.NamespacedName)
+```
+
+### Change Logger to use flag options
+
+In the `main.go` file replace: 
+
+```go
+flag.Parse()
+
+ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
+```
+
+With:
+
+```go
+opts := zap.Options{
+		Development: true,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts))) 
+```
+
+### Rename the manager flags
+
+The manager flags `--metrics-addr` and `enable-leader-election` was renamed for `--metrics-bind-address` and `--leader-elect` to be more aligned with k8s Kubernetes Components. More info: [#1839][issue-1893]. 
+
+In your `main.go` file replace: 
+
+
+```go
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
+```
+
+With: 
+
+```go
+func main() {
+	var metricsAddr string
+	var enableLeaderElection bool
+	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
```

Comment:
```suggestion
	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
		"Enable leader election for controller manager. "+
			"Enabling this will ensure there is only one active controller manager.")
```
Maybe the above code snippet should be this?
I created a new project from kubebuilder master branch and got the new code from that.